backend/src/main/java/com/github/jekattack/cornergame/game/quests/QuestService.java

```diff
@@ -5,27 +5,44 @@
 import com.github.jekattack.cornergame.game.gamedata.CGUserGameDataService;
 import com.github.jekattack.cornergame.game.gamedata.questItem.QuestItem;
 import com.github.jekattack.cornergame.game.gamedata.questItem.QuestStatus;
+import com.github.jekattack.cornergame.game.visits.Visit;
+import com.github.jekattack.cornergame.game.visits.VisitObserver;
+import com.github.jekattack.cornergame.game.visits.VisitRepository;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
 
 import java.time.Instant;
-import java.time.temporal.ChronoUnit;
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.List;
-import java.util.NoSuchElementException;
-import java.util.stream.Collectors;
+import java.util.*;
 
 @Service
 @RequiredArgsConstructor
 @Slf4j
-public class QuestService {
+public class QuestService implements VisitObserver {
 
     private final QuestRepository questRepository;
-    private final CGUserGameDataRespository cgUserGameDataRespository;
+    private final VisitRepository visitRepository;
     private final CGUserGameDataService cgUserGameDataService;
+    private final List<QuestObserver> questObservers;
+
```

Comment:
if you serve the data you fetch from VisitRepository with the onVisitCreated and solve that problem through observer, you wont need the import anymore, which seems nice